src/github.com/couchbase/sync_gateway/db/index_changes.go

```diff
@@ -168,13 +117,19 @@ func (db *Database) VectorMultiChangesFeed(chans base.Set, options ChangesOption
 				if minEntry.Seq.TriggeredBy == 0 {
 					// Update the cumulative clock, and stick it on the entry.
 					cumulativeClock.SetMaxSequence(minEntry.Seq.vbNo, minEntry.Seq.Seq)
-					// TODO: Potential enhancement - for first iteration loop, always set hash on last entry.
-					continuousLastHash = db.calculateHashWhenNeeded(
+					// Force new hash generation for non-continuous changes feeds if this is the last entry to be sent - either
+					// because there are no more entries in the channel feeds, or we're going to hit the limit.
+					forceHash := false
+					if options.Continuous == false && (nextEntry == nil || options.Limit == 1) {
+						forceHash = true
```

Comment:
The goal is to identify when we're about to break out of the inner for loop.  

nextEntry==nil is the primary case (since Limit is an optional parameter), and represents that there aren't any more entries in ANY channel.  It's the same value that's being checked as minEntry==nil in the next loop iteration (above), that triggers the break that ends the changes loop iteration.  

If options.Limit==1 we're about to break out of the loop below (line 209).